src/core/base.ts

```diff
@@ -96,7 +97,9 @@ export class Base implements InterfaceBase {
   }
   private async auth(password: string) {
     try {
-      return await this.command("AUTH", password);
+      const authResponse = await this.command("AUTH", password);
+      if (authResponse instanceof Error) { throw authResponse; }
+      return authResponse;
```

Comment:
One open task with this pull-request is my handling of errors. Currently, `protocol.ts` doesn't throw errors directly (it just returns an `Error` object), so it's up to the caller to check if this is an `Error` object. The primary reason for this is: commands which can return multiple results, it's possible for partial success and thus I didn't want to throw an `Error` if 9 out of 10 commands (in a `MULTI` for example) succeeded, but one failed.